security/acme-client/src/opnsense/scripts/OPNsense/AcmeClient/certhelper.php

```diff
@@ -327,9 +327,17 @@ function eval_optional_acme_args()
     $acme_args[] = $configObj->OPNsense->AcmeClient->settings->environment == "stg" ? "--staging" : null;
     $acme_args[] = isset($options["S"]) ? "--staging" : null; // for debug purpose
 
-    // Set log level
-    $acme_args[] = $configObj->OPNsense->AcmeClient->settings->logLevel == "normal" ? "--log-level 1" : "--log-level 2";
-    $acme_args[] = $configObj->OPNsense->AcmeClient->settings->logLevel == "debug" ? "--debug" : null;
+    // Set log level - only one level chosen from dropdown
+    switch($configObj->OPNsense->AcmeClient->settings->logLevel) {
+        case "extended":
+            $acme_args[] = "--log-level 2";
+        case "debug":
+            $acme_args[] = "--debug";
+        case "dnsapi":
+            $acme_args[] = "--debug 2";
+        default:
+            $acme_args[] = "--log-level 1";
+     }
```

Comment:
Same here, please move this change to a new PR to separate concerns :)

Reply:
Agreed. As stated in a comment to Frank Wall there is probably a better way of doing this. That is have a separate selection come if debug is selected and then set the debug level from there. Seems that the debug levels are somewhat open ended. Looks like a level 3 is also possible. Playing with debugging I saw that the dnsapi scripts __debug_ lines only show up at _--debug 2_. However you can declare a __debug2_ in dnsapi scripts. Think those debug lines won't be logged unless you select _--debug 3_. I'll give it a test and post back.